### ISSUE TITLE
Issue with PropertyDeclaration#setType

### ISSUE THREAD

**Issue by sfrooster** [dsherret/ts-simple-ast#53]
Processing the following code:
```typescript
        export class ReservationModified extends Message {
            public static readonly Version: string = "1.0";   // version to pass to EDAProperties
            public static readonly TypeID = "Events.CRUD.ReservationModified";

            public get Key(): number[] {
                return numToByteArray(this.ReservationID!);
            }

            // Some values are optional, see https://wiki.otcorp.opentable.com/display/CP/Reservation+Event%2C+Version+1
            constructor();
            constructor(ReservationID: number, UpdatedState: ReservationEventState, PreviousState: ReservationEventState, EventDateUTC: moment.Moment, HighWaterMark: number)
            constructor(
                public readonly ReservationID: number = 0,
                public readonly UpdatedState?: ReservationEventState,
                public readonly PreviousState?: ReservationEventState,
                public readonly EventDateUTC: moment.Moment = moment.utc(),
                public readonly HighWaterMark?: number) {
                super();
            }
        }
```
the following code chokes at smember.setType:
```typescript
    let staticMembers = classDec.getStaticMembers();

    staticMembers.forEach(smember => {
        if (smember instanceof PropertyDeclaration) {
            switch (smember.getName()) {
                case "TypeID":
                    if (smember.getTypeNode() === undefined) {
                        smember.setType("string");
                        log("added type to TypeID", 2);
                        changeTracker.recordChange();
                    }
```
specifically here where currentNodeChildren.next() is never checked for "done === true"
```typescript
function replaceStraight(sourceFile, replacePos, replaceLength, newText) {
    const compilerFactory = sourceFile.global.compilerFactory;
    const currentText = sourceFile.getFullText();
    const newFileText = currentText.substring(0, replacePos) + newText + currentText.substring(replacePos + replaceLength);
    const tempSourceFile = compilerFactory.createTempSourceFileFromText(newFileText, sourceFile.getFilePath());
    const endPos = replacePos + newText.length;
    const removedNodes = [];
    handleNode(sourceFile, tempSourceFile);
    removedNodes.forEach(n => n.dispose());
    function handleNode(currentNode, newNode) {
        /* istanbul ignore if */
        if (currentNode.getKind() !== newNode.getKind())
            throw new errors.InvalidOperationError(getInsertErrorMessageText_1.getInsertErrorMessageText("Error inserting straight.", currentNode, newNode));
        const currentNodeChildren = currentNode.getChildrenIterator();
        let currentNodeChild;
        for (const newNodeChild of newNode.getChildrenIterator()) {
            // todo: is getStart slow? Maybe something could be added or changed here for performance reasons
            const newNodeChildStart = newNodeChild.getStart();
            if (newNodeChildStart >= replacePos && newNodeChildStart < endPos)
                continue;
            let currentNodeChildStart;
            currentNodeChild = undefined;
            do {
                if (currentNodeChild != null)
                    removedNodes.push(currentNodeChild);
                currentNodeChild = currentNodeChildren.next().value;
                currentNodeChildStart = currentNodeChild.getStart();
            } while (currentNodeChild.getKind() === ts.SyntaxKind.TypeReference ||
                replaceLength > 0 && currentNodeChildStart >= replacePos && currentNodeChildStart < replacePos + replaceLength);
            handleNode(currentNodeChild, newNodeChild);
        }
        compilerFactory.replaceCompilerNode(currentNode, newNode.compilerNode);
    }
}
```

**Comment by dsherret** [comment]
Hey Michel, this should be fixed in 0.48.1. Please reopen this issue if not.

That replaceStraight function is going to be removed soon... I'm working on standardizing and cleaning up the underlying manipulation functions in #47.

**Comment by sfrooster** [comment]
Yep - 48.1 works!
